src/composables/websocket.ts

```diff
@@ -0,0 +1,155 @@
+    })));
+    channels.value[channelName] = true;
+  }
+
+  /**
+   * Handle websocket message event.
+   *
+   * @param {MessageEvent} message - The message event object.
+   */
+  function handleWebsocketMessage(message: MessageEvent) {
+    if (!message.data) return;
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers.value[data.subscription as WebSocketChannelTypes]
+      ) return;
+      // Call all subscribers for the channel
+      subscribers.value[data.subscription as WebSocketChannelTypes].forEach(
+        (subscriberCb: any) => subscriberCb(data.payload),
+      );
+    } catch (err) {
+      //
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  async function subscribeToWebsocketChannels() {
+    if (!wsClient.value) {
+      return;
+    }
+    // Add an event listener for messages
+    wsClient.value.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    Object.keys(channels.value).forEach(subscribeToWebsocketChannel as any);
+  }
+
+  /**
+   * UnSubscribes websocket channels.
+   */
+  async function unsubscribeFromWebsocketChannels() {
+    if (!wsClient.value) {
+      return;
+    }
+
+    // Remove event listener for messages
+    wsClient.value.removeEventListener('message', handleWebsocketMessage);
+
+    // UnSubscribe all channels
+    Object.keys(channels.value).forEach((channelName: string) => {
+      channels.value[channelName as WebSocketChannelTypes] = false;
+    });
+  }
+
+  /**
+   * This function creates a new WebSocket client and attempts to connect to the specified server.
+   * @param {string} url - The URL of the WebSocket server to connect to
+   */
+  async function connectToWebsocketServer(url: string) {
+    if (wsClient.value) {
+      await unsubscribeFromWebsocketChannels();
+
+      if (isConnected.value) {
+        wsClient.value.close();
+      }
+    }
+
+    isConnected.value = false;
+
+    wsClient.value = new WebSocket(url);
+
+    wsClient.value.onopen = () => {
+      isConnected.value = true;
+      subscribeToWebsocketChannels();
+    };
+
+    wsClient.value.onclose = () => {
+      isConnected.value = false;
+    };
+  }
+
+  /**
+   * Subscribes to a specific websocket channel
+   *
+   * @param {WebSocketChannelTypes} channel - The websocket channel to subscribe to.
+   * @param {Function} [callback=(payload: any) => payload]
+   *    - The function to be called when a message is received on the specified channel.
+   */
+  function channelSubscribe(
+    channel: WebSocketChannelTypes,
+    // eslint-disable-next-line no-unused-vars
+    cb: (payload: any) => void = (payload: any) => null,
+  ) {
+    // subscribe to channel
+    subscribeToWebsocketChannel(channel);
+    // register the subscriber
+    if (!subscribers.value[channel]) {
+      subscribers.value[channel] = [];
+    }
+    subscribers.value[channel].push(cb);
```

Comment:
There is a default value for the `cb` which is an empty function. Is this really required to have empty functions added to this array? Maybe we should have the default value set to null and add the callback only if it's not null?